script/BlockBuilderPolicy.s.sol

```diff
@@ -11,15 +12,27 @@ import {FlashtestationRegistry} from "../src/FlashtestationRegistry.sol";
 /// @dev the FLASHTESTATION_REGISTRY_ADDRESS env var is the address of the contract deployed by the FlashtestationRegistryScript
 /// @dev the OWNER_BLOCK_BUILDER_POLICY env var is the address that can add and remove workloads from the policy
 contract BlockBuilderPolicyScript is Script {
-    BlockBuilderPolicy public policy;
-    FlashtestationRegistry public registry;
-
     function setUp() public {}
 
     function run() public {
         vm.startBroadcast();
-        policy = new BlockBuilderPolicy(
-            vm.envAddress("FLASHTESTATION_REGISTRY_ADDRESS"), vm.envAddress("OWNER_BLOCK_BUILDER_POLICY")
+        // this is the address that stores all of the TEE-controlled addresses and their associated workloadIds
+        address registry = vm.envAddress("FLASHTESTATION_REGISTRY_ADDRESS");
+        console.log("registry", registry);
+        // do some safety checks to make sure this is indeed a registry contract
+        require(registry != address(0), "FLASHTESTATION_REGISTRY_ADDRESS address is 0x0");
+        FlashtestationRegistry registryContract = FlashtestationRegistry(registry);
+        require(
+            registryContract.owner() == vm.envAddress("FLASHTESTATION_REGISTRY_OWNER"),
```

Comment:
it seems like the registry contract is already deployed at this point? why do we need to asset the registry owner here?

Reply:
it's just a sanity check to make sure that, before we go ahead and make the script's function call, we ensure we're working with the registry contract that we think we're working with.